litho-it/src/test/java/com/facebook/litho/specmodels/generator/BuilderGeneratorTest.java

```diff
@@ -186,4 +196,216 @@ public void testGenerate() {
                 + "  }\n"
                 + "}\n");
   }
+
+  @Test
+  public void testResTypeWithVarArgs() {
+    TypeSpecDataHolder dataHolder = BuilderGenerator.generate(mSpecModel1);
+    assertThat(dataHolder.getTypeSpecs()).hasSize(1);
+    assertThat(dataHolder.getTypeSpecs().get(0).toString())
+        .isEqualTo(
+            "public static class Builder extends com.facebook.litho.Component.Builder<com.facebook.litho.specmodels.generator.BuilderGeneratorTest.TestResTypeWithVarArgs, Builder> {\n" +
+            "  private static final java.lang.String[] REQUIRED_PROPS_NAMES = new String[] {\"arg0\"};\n" +
+            "\n" +
+            "  private static final int REQUIRED_PROPS_COUNT = 1;\n" +
+            "\n" +
+            "  TestResTypeWithVarArgsImpl mTestResTypeWithVarArgsImpl;\n" +
+            "\n" +
+            "  com.facebook.litho.ComponentContext mContext;\n" +
+            "\n" +
+            "  private java.util.BitSet mRequired = new java.util.BitSet(REQUIRED_PROPS_COUNT);\n" +
+            "\n" +
+            "  private void init(com.facebook.litho.ComponentContext context, int defStyleAttr, int defStyleRes,\n" +
+            "      TestResTypeWithVarArgsImpl testResTypeWithVarArgsImpl) {\n" +
+            "    super.init(context, defStyleAttr, defStyleRes, testResTypeWithVarArgsImpl);\n" +
+            "    mTestResTypeWithVarArgsImpl = testResTypeWithVarArgsImpl;\n" +
+            "    mContext = context;\n" +
+            "    mRequired.clear();\n" +
+            "  }\n" +
+            "\n" +
+            "  public Builder test(java.lang.Float test) {\n" +
```

Comment:
Sorry, I may just be missing something here, but is this needed? Seems to be the same as below apart from the boxing.

Reply:
I was thinking of this neither. To be honest, I don't know why should I need a `Float` method with some `resType`s, But maybe when you declaring `varArgs` with `List<Float>` generate a method with `Float` is a right behaviour. I am not sure about that or I can just remove it.

Reply:
Since `DIMEN_SIZE` `DIMEN_OFFSET` and `DIMEN_TEXT` do not add regularBuilder so I figure out that the `Float` method is indeed redundant. I move the `varArgBuilder` block into `ResType.None` and add a new `regularBuilders` to handle both `regularBuilder` and `varArgBuilder` for `non-dimen` `resType` so it's just as same as before in concept.

Reply:
Awesome!